### PR TITLE
Typescript && GPR Package

### PR DESCRIPTION
**Pull request by iandvt**
This will allow the repository to be consumed using `yarn/npm install @formidablelabs/dogs`. This project also serves as a reference for publishing packages to GPR using semantic-release.

Github Action - [Link](https://github.com/FormidableLabs/dogs/actions/runs/309203107/workflow)

### Sample Release Notes Below
# 1.0.0 (2020-10-16)

### Features

    * typescript && gpr package (a3c8363)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
   },
-  "author": "",
-  "license": "MIT",
-  "bugs": {
-    "url": "https://github.com/FormidableLabs/dogs/issues"
+  "publishConfig": {
+    "registry": "https://npm.pkg.github.com"
+  },
+  "scripts": {
+    "build": "rimraf dist/ && tsc"
```
Comment (ryan-roemer) [inline package.json:14]: Do we want to add a `preversion` workflow to make sure that everything appropriate is released? A la https://github.com/FormidableLabs/trace-deps/blob/master/CONTRIBUTING.md#releasing-a-new-version-to-npm

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Mac
+.DS_STORE
+
+# Node
+node_modules/
+dist/
+.npmrc
```
Comment (ryan-roemer) [inline .gitignore:7]: Right now `dist/` won't be published. We need something like https://github.com/FormidableLabs/trace-deps/blob/master/.npmignore (which is a reverse "include only what we want" version)

Comment (ryan-roemer) [reply .gitignore:7]: The `npm pack` command is great for giving you a preview of what will publish for real.

File: package.json (hunk 1 above)
Comment (iandvt) [reply package.json:14]: What will be published is included in the [GithHub workflow](https://github.com/FormidableLabs/dogs/blob/typescript-package/.github/workflows/gpr-publish.yaml#L22) via `npm publish --dry-run`. 

[Example](https://github.com/FormidableLabs/dogs/runs/1260574891?check_suite_focus=true#step:6:9)

File: .gitignore (hunk 2 above)
Comment (iandvt) [reply .gitignore:7]: `/dist` is generated in the [GitHub workflow](https://github.com/FormidableLabs/dogs/blob/typescript-package/.github/workflows/gpr-publish.yaml#L20) via `npm run build`

[Example](https://github.com/FormidableLabs/dogs/runs/1260574891?check_suite_focus=true#step:5:9)

File: .github/workflows/gpr-publish.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout code
+        uses: actions/checkout@v2
+      -
+        name: Use Node.js 12
```
Comment (ryan-roemer) [inline .github/workflows/gpr-publish.yaml]: Can we switch to 14?

File: .github/workflows/gpr-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - 
+        run: npm install -g yarn 
+      -
+        run: yarn install --frozen-lockfile
+      - 
+        run: npm run build
```
Comment (ryan-roemer) [inline .github/workflows/gpr-publish.yaml]: Minor: Let's do `yarn run build` and save `npm` for just the publish commands.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,8 @@
 # dogs
 Formidable Dogs
+
+## Contributing
+This repository is configured using [semantic-release](https://github.com/semantic-release/semantic-release) which automates the whole package release workflow including: determining the next version number, generating the release notes and publishing the package.
+
```
Comment (ryan-roemer) [inline README.md:6]: Minor: Remove extra newline here.

File: package.json [hunk 6]
```diff
@@ -1,19 +1,40 @@
...
+    "@commitlint/cli": "^11.0.0",
+    "@commitlint/config-conventional": "^11.0.0",
+    "husky": "^4.3.0",
+    "rimraf": "^3.0.2",
+    "semantic-release": "^17.2.1",
+    "typescript": "^4.0.3"
+  },
+  "commitlint": {
+    "extends": [
+      "@commitlint/config-conventional"
```
Comment (ryan-roemer) [inline package.json:30]: Are there some restrictions for commit messages that we should add to documentation?

File: README.md [hunk 7]
```diff
@@ -1,2 +1,8 @@
 # dogs
 Formidable Dogs
+
+## Contributing
+This repository is configured using [semantic-release](https://github.com/semantic-release/semantic-release) which automates the whole package release workflow including: determining the next version number, generating the release notes and publishing the package.
```
Comment (ryan-roemer) [inline README.md:5]: Can we add a simple example of how "inputs" translate to a version number choice and the resulting published package?

**Pull request merged**